virttest/cartesian_config.py

```diff
@@ -947,7 +947,8 @@ def apply_to_dict(self, d):
         exp = re.compile("%s$" % self.name)
         keys_to_del = collections.deque()
         for key in d:
-            if key not in _reserved_keys and exp.match(key):
+            keystr = "".join(key) if isinstance(key, tuple) else key
```

Comment:
Can we ensure that element in tuple is str instance, otherwise "".join(key) will throw TypeError?
```
>>> my_tuple = (1, "Hello", 3.4)
>>> "".join(my_tuple)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: sequence item 0: expected str instance, int found

```

Reply:
Thanks @chunfuwen for the review. The problem here is that:

1) This is the way tuples are handled for various other operators and was simply missing for the del operator.

2) There are already assumptions in place that all parameters are either strings or tuples of strings. So we might benefit from a central location to validate this assumption if such is no longer present.

What do you think?

Reply:
thank for explanation

Reply:
In any case, I will note your comment down for the central location for a further PR unless somebody else goes before me :heavy_check_mark: